### PR TITLE
Allow token/ip in Link

### PR DESCRIPTION
**Pull request by retog**
This allows to get a QR Code that allows guests to control the light

https://youtu.be/89F4GWq-YdE

### CODE REVIEW

File: js/login.js [hunk 1]
```diff
@@ -5,7 +5,22 @@ document.getElementById("manuallyIpBtn").addEventListener("click", manuallyIpNew
...
+
+function checkQueryArgs() {
+    const params = new URLSearchParams(window.location.search);
+    const token = params.get("token")
+    const ip = params.get("ip")
+    if (token && ip) {
+        const acc = {token,ip};
+        return connectionGood(acc).then(() => {
+            localStorage.setItem('hueAcc', JSON.stringify(acc));
+            window.location.search = "";
```
Comment (JakobJFL) [inline js/login.js]: I don't understand why the `window.location.search = "";` is necessary and if it's there I can't login using the URL parameters. So I will suggest removing it if it doesn't serve any other purpose.

**Review by JakobJFL — CHANGES_REQUESTED** [review]
Thank you so much for your contribution.

I would love to add this to the project I think it is an interesting idea.
I tried using it and had a slight issue that I added a comment on. 
Otherwise, I think it looks fine :).

File: js/login.js (hunk 1 above)
Comment (retog) [reply js/login.js]: It is not strictly necessary. It's for the esthetics of the URL, preventing accidental reconnecting after explicitly disconnecting, and to not show token and network information when sharing the screen.

**Comment by retog** [comment]
I suggest keeping it as is. Having the token both in the (bookmarkable) URL as well as in localStorage is a duplication that can lead to confusion, So because of this the token in the URL is a one-time thing to set it, after which the token Is revived from the URL.

An alternative approach would be not to use localStorage and always have the token in the URL. This would allow bookmarking and using multiple controllers for different bridges in parallel,

File: js/login.js (hunk 1 above)
Comment (JakobJFL) [reply js/login.js]: That's a fair point but my issue is that it does not work in my Chrome browser if line 18 is not commented out...

Comment (retog) [reply js/login.js]: Wondering if this is caused by a refresh it triggers. If so, this could be the solution: https://stackoverflow.com/a/28609299/1455912

Comment (retog) [reply js/login.js]: Let me know if it works now

Comment (JakobJFL) [reply js/login.js]: Looks to be working fine now. Thanks for your contribution!

**Review by JakobJFL — APPROVED** [review]

**Pull request merged**